nx/guides/advanced/automatic_differentiation.livemd

```diff
@@ -0,0 +1,182 @@
+# Automatic Differentation
+
+```elixir
+Mix.install([
+  {:nx, "~> 0.7"}
+])
+```
+
+## What is Function Differentiation?
```

Comment:
nitpick: let's make sure we are consistent with titles and if we use "What is function differentiation?" or "What is Function Differentiation?". FWIW, in the Elixir repo itself, we use the former, but we can follow another convention here.